azurerm/resources.go

```diff
@@ -0,0 +1,135 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/cycloidio/terracognita/provider"
+	"github.com/cycloidio/terracognita/tag"
+)
+
+// ResourceType is the type used to define all the Resources
+// from the Provider
+type ResourceType int
+
+//go:generate enumer -type ResourceType -addprefix azurerm_ -transform snake -linecomment
+const (
+	ResourceGroup          ResourceType = iota
+	VirtualMachine         ResourceType = iota
+	VirtualNetwork         ResourceType = iota
+	Subnet                 ResourceType = iota
+	NetworkInterface       ResourceType = iota
+	NetworkSecurityGroup   ResourceType = iota
+	VirtualMachineScaleSet ResourceType = iota
+)
```

Comment:
correct, but we should use implicit repetition to be consistent with other providers.
```suggestion
const (
	ResourceGroup          ResourceType = iota
	VirtualMachine
	VirtualNetwork
	Subnet
	NetworkInterface
	NetworkSecurityGroup
	VirtualMachineScaleSet
)
```